tokio-stream/src/wrappers/broadcast.rs

```diff
@@ -0,0 +1,38 @@
+use crate::Stream;
+use async_stream::try_stream;
+use std::pin::Pin;
+use tokio::sync::broadcast::error::RecvError;
+use tokio::sync::broadcast::Receiver;
+
+use std::task::{Context, Poll};
+
+/// A wrapper around [`Receiver`] that implements [`Stream`]. Achieved by using the [`async-stream`] crate.
+///
+/// [`Receiver`]: struct@tokio::sync::broadcast::Receiver
+/// [`Stream`]: trait@crate::Stream
+/// [`async-stream`]: https://docs.rs/async-stream
+#[derive(Debug)]
+pub struct BroadcastStream<T: Clone> {
+    inner: Pin<Box<dyn Stream<Item = Result<T, RecvError>> + Send + Sync >>,
+}
+
+impl<T: Clone + Unpin + 'static + Send + Sync> BroadcastStream<T> {
+    /// Create a new `BroadcastStream`.
+    pub fn new(mut rx: Receiver<T>) -> Self {
+        let stream = try_stream! {
+            loop {
+                let item = rx.recv().await?;
+                yield item;
+            }
+        };
```

Comment:
I think it's simpler to write like this. Then the shutdown condition is clearer. That said, it is equivalent.
```suggestion
        let stream = stream! {
            while let Some(item) = rx.recv().await {
                yield item;
            }
        };
```

Reply:
Oh? would it be exactly the same or? Isn't the `recv()` return a Result? 🤔  So for your change it would need to be `while let Ok(item)` ? And the error result will be just ignored?